src/harness/test_harness_objects.py

```diff
@@ -0,0 +1,324 @@
+                                               self.ssl_cert, self.ssl_key)['grantResponse']
+
+    # Make one CBSD object per successful grant request
+    for index, grant_request in enumerate(grant_requests):
+      grant_request["cbsdId"] = cbsd_ids[index]
+      if grant_responses[index]['response']['responseCode'] == 0:
+        cbsd_object = Cbsd(cbsd_ids[index], registration_requests[index],
+                           [grant_responses[index]['grantId']], [grant_request])
+        self.cbsd_objects[cbsd_ids[index]] = cbsd_object
+
+  def getCbsdObjectById(self, cbsd_id):
+      return self.cbsd_objects[cbsd_id]
+
+  def heartbeatRequestForAllActiveGrants(self):
+    """Performs heartbeat request for all grants of CBSD devices at once.
+
+    Returns:
+      heartbeat_requests : list of heart beat requests which are constructed for all the grants
+      by the caller
+      heartbeat_responses: list of heart beat response
+    """
+    # Concatenate all the heartbeat request of CBSD into single request
+    heartbeat_requests = []
+    for cbsd_object_item in self.cbsd_objects.values():
+      heartbeat_requests.extend(cbsd_object_item.constructHeartbeatRequestForAllActiveGrants())
+
+    if len(heartbeat_requests):
+      heartbeat_requests_wrapped = {
+        'heartbeatRequest': heartbeat_requests
+      }
+      print heartbeat_requests_wrapped
+      # Perform heartbeat requests
+      heartbeat_responses = self.testcase._sas.Heartbeat(
+        heartbeat_requests_wrapped, self.ssl_cert, self.ssl_key)['heartbeatResponse']
+
+      for heartbeat_response in heartbeat_responses:
+        cbsd_object = self.cbsd_objects[heartbeat_response['cbsdId']]
+        grant_object = cbsd_object.grant_objects[heartbeat_response['grantId']]
+
+        # Update grant state based on the heartbeat response
+        self._mapResponseCodeToGrantState(heartbeat_response['response']['responseCode'],
+                                         grant_object)
+
+      return heartbeat_requests, heartbeat_responses
+
+  def _mapResponseCodeToGrantState(self, heartbeat_response_code, grant_object):
+    """Maps the heartbeat response and update the grant state."""
+    if heartbeat_response_code == Grant.ResponseCodes.SUCCESS.value:
+      grant_object.authorized_in_last_heartbeat = True
+    elif heartbeat_response_code == Grant.ResponseCodes.TERMINATED_GRANT.value:
+      grant_object.is_terminated = True
+    elif heartbeat_response_code == Grant.ResponseCodes.SUSPENDED_GRANT.value:
+      grant_object.authorized_in_last_heartbeat = False
+    elif heartbeat_response_code == Grant.ResponseCodes.UNSYNC_OP_PARAM.value:
+      grant_object.is_terminated = True
+      grant_object.authorized_in_last_heartbeat = False
+      logging.error('Received UNSYNC_OP_PARAM code in Heartbeat Response')
+    else:
+      logging.error('Unknown code=%s received in Heartbeat Response',heartbeat_response_code)
+
+  def _constructGrantRequests(self,heartbeat_request):
```

Comment:
The name should be singular since you are constructing only one.

Reply:
Corrected   as per suggestion.